packages/test-sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/TestApp.kt

```diff
@@ -50,6 +51,15 @@ val TEST_APP_CLUSTER_NAME = SyncServerConfig.clusterName
 val TEST_SERVER_BASE_URL = SyncServerConfig.url
 const val DEFAULT_PASSWORD = "password1234"
 
+// Expose a try-with-resource pattern for Test Apps
+inline fun App.use(action: (App) -> Unit) {
```

Comment:
Given that we also forget to wrap apps in this ourselves, should we then consider to rework the `TestApp`-constructor to a factory method with a block that always executed in a try-finally block 🤔

Reply:
I guess we can't do this because the app is mostly setup in `@BeforeTest` and `use`d in the actual `@Test`

Reply:
I created https://github.com/realm/realm-kotlin/issues/1496 to track this